### PR TITLE
fix: use 0 as default ns for old kernel

### PR DESCRIPTION
**Pull request by eddietanshuo**
### This PR is for:
[procfs](https://docs.rs/procfs/latest/procfs/process/struct.Process.html#method.namespaces) `/proc/{pid}/ns` only works for linux 3.0+, use `0` as default netns identify for old kernel.

### CODE REVIEW

File: agent/src/platform/platform_synchronizer/linux_socket.rs [hunk 1]
```diff
@@ -308,13 +308,22 @@ fn is_zero_addr(addr: &SocketAddr) -> bool {
 }
 
 fn get_proc_netns(proc: &Process) -> Result<u64, ProcError> {
-    if let Some(netns) = proc.namespaces()?.get(&OsString::from("net")) {
-        Ok(netns.identifier)
-    } else {
-        Err(ProcError::Other(format!(
-            "pid {} get net ns fail",
-            proc.pid
-        )))
```
Comment (sharang) [inline agent/src/platform/platform_synchronizer/linux_socket.rs:317]: 感觉可以简化一些

```
    if let Some(netns) = proc.namespaces()?.get(&OsString::from("net")) {
        Ok(netns.identifier)
    } else {
        // works with linux 3.0+ kernel only,
        // refer to this [commit](https://github.com/torvalds/linux/commit/6b4e306aa3dc94a0545eb9279475b1ab6209a31f)
        // return 0 as default netns identify as a walkaround
        Ok(0)
    }
```

Comment (eddietanshuo) [reply agent/src/platform/platform_synchronizer/linux_socket.rs:317]: 上面`?` 应该会直接return一个Err 走不到else分支

**Review by sharang — DISMISSED** [review]

**Comment by sharang** [comment]
@eddietanshuo 还需要点一下 CLA

**Comment by eddietanshuo** [comment]
我简化下代码，保证尽量优雅，稍后我测试下

**Review by sharang — APPROVED** [review]

**Review by sharang — APPROVED** [review]

**Pull request merged**